src/destructors.md

```diff
@@ -126,6 +126,15 @@ r[destructors.scope.nesting.match-arm]
 * The parent of the expression after the `=>` in a `match` expression is the
   scope of the arm that it's in.
 
+r[destructors.scope.nesting.match-guard-pattern]
+* The parent of an `if let` guard pattern is the scope of the guard expression.
```

Comment:
I'm struggling with this one. Isn't the parent of both the scrutinee and the bindings of an `if let` guard the entire arm?  I'm not able to find a way that it is different from destructors.scope.nesting.match-guard. 

For example, in:
```rust
match x {
    Some(a) if let [one] = some_expression => {}
    _ => {}
}
```

I would expect the parent of both `one` and `some_expression` to be the match arm (and drop *after* the arm body).

Can you clarify this for me?